pkg/restic/commands.go

```diff
@@ -98,82 +87,121 @@ func CreateBackup(ctx context.Context, globalOpts *GlobalOptions, backupOpts *Ba
 	args = cli.StructToCLI(globalOpts)
 	args = append(args, cli.StructToCLI(backupOpts)...)
 
-	cmd := cli.CommandType{
-		Binary:  binary,
-		Command: "backup",
-		Args:    args,
-	}
+	cmd := newCommand("backup", args...)
+
 	out, err = cli.RunWithTimeout(ctx, cmd, cmdTimeout)
 	if err != nil {
 		return BackupResult{}, out, err
 	}
-
-	backupRes, err := parseSnapshotOut(fmt.Sprintf("%s", out))
+	var response Response
+	err = response.ReadResponseFromLines(out)
+	if err != nil {
+		return BackupResult{}, out, err
+	}
+	backupRes, err := parseSnapshotOut(response)
 	if err != nil {
 		return backupRes, out, err
 	}
 	return backupRes, nil, nil
 }
 
+func (response *Response) ReadResponseFromLines(data []byte) error {
```

Comment:
I would convert this to a non-receiver function like `NewResponseFromResticCLIOutput(data []byte) (Response, error)`